move_base/src/move_base.cpp

```diff
@@ -681,7 +681,13 @@ namespace move_base {
       if(as_->isPreemptRequested()){
         if(as_->isNewGoalAvailable()){
           //if we're active and a new goal is available, we'll accept it, but we won't shut anything down
-          move_base_msgs::MoveBaseGoal new_goal = *as_->acceptNewGoal();
+          boost::shared_ptr<const move_base_msgs::MoveBaseGoal_<std::allocator<void> > > new_goalPtr = as_->acceptNewGoal();
```

Comment:
Question: 
 Is this pointer can be deleted by delete()? 
 Is there the release function corresponding to acceptNewGoal()?

Reply:
The type of acceptNewGoal is boost :: shared_ptr as shown below.
```
boost::shared_ptr<const move_base_msgs::MoveBaseGoal_<std::allocator<void> > > new_goalPtr = as_->acceptNewGoal();
```
The instance as_ is a MoveBaseActionServer class.
```
as_ = new MoveBaseActionServer(ros::NodeHandle(), "move_base", boost::bind(&MoveBase::executeCb, this, _1), false);

move_base/include/move_base/move_base.h:64:  typedef actionlib::SimpleActionServer<move_base_msgs::MoveBaseAction> MoveBaseActionServer;
```
MoveBaseActionServer class corresponds to SimpleActionServer.
```
server/simple_action_server.h:139:  boost::shared_ptr<const Goal> acceptNewGoal();
```
The return value of acceptNewGoal returns "boost :: shared_ptr".
Therefore, there is no explicit delete processing.

Reply:
OK.